experimental/devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package bmp180 controls a Bosch BMP180 device over I²C.
+//
+// Datasheet
+//
+// The official data sheet can be found here:
+//
+// https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMP180-DS000-121.pdf
+//
+// The font the official datasheet on page 15 is unreadable, a copy with
+// readable text can be found here:
+//
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+//
+// Notes on the datasheet
+//
+// The results of the calculations in the algorithm on page 15 are partly
+// wrong. It looks like the original authors used non-integer calculations and
+// some nubers were rounded. Take the results of the calculations with a grain
+// of salt.
+package bmp180
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/devices"
+)
+
+// Oversampling affects how much time is taken to measure pressure.
+type Oversampling uint8
+
+const (
+	// Possible oversampling values
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+
+	// bit offsets in regCtrlMeas
+	ctrlMeasurementControlShift = 0
+	ctrlStartConversionShift    = 5
+	ctrlOversamplingShift       = 6
+
+	// commands
+	cmdStartTempConv     uint8 = (1 << ctrlStartConversionShift) | (0x0E << ctrlMeasurementControlShift)
+	cmdStartPressureConv uint8 = (1 << ctrlStartConversionShift) | (0x14 << ctrlMeasurementControlShift)
+
+	chipAddress = 0x77 // the address of a BMP180 is fixed at 0x77
+	chipID      = 0x55 // contents of the ID register, always 0x55
+
+	// registers
+	regChipID           = 0xD0 // register contains the chip id
+	regCalibrationStart = 0xAA // first calibration register address
+	regSoftReset        = 0xE0 // soft reset register
+	regCtrlMeas         = 0xF4 // control measurement
+	regOut              = 0xF6 // 3 bytes register with measurement data
+
+	softResetValue = 0xB6
+
+	tempConvTime = 4500 * time.Microsecond // maximum conversion time for temperature
+)
+
+// maximum conversion time for pressure
+var pressureConvTime = [...]time.Duration{
+	4500 * time.Microsecond,
+	7500 * time.Microsecond,
+	13500 * time.Microsecond,
+	25500 * time.Microsecond,
+}
+
+// Dev is a handle to a bmp180.
+type Dev struct {
+	dev *mmr.Dev8
```

Comment:
I just realized you don't need it to be a pointer, please inline instead of allocating on the heap. Last comment I promise.